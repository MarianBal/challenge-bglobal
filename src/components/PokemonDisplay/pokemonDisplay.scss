@import '../../resources/mixin.scss';

.display-container {
  width: 70%;
  padding: 15px 0;
  @include flexbox(row, center, center) flex-wrap: wrap;
}
