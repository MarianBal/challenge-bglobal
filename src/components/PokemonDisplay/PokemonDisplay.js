import React, { useState, useEffect } from 'react';

import Card from '../Card/Card';
import apiCall from '../../utils/apiCall';
import './pokemonDisplay.scss';

const url = 'https://pokeapi.co/api/v2/pokemon?limit=40&offset=0';

const PokemonDisplay = () => {
  const [results, setResults] = useState([]);

  useEffect(() => apiCall(url, setResults), []);
  return (
    <div className='display-container'>
      {results.length && results.map(pokemon => <Card pokemon={pokemon} />)}
    </div>
  );
};

export default PokemonDisplay;
