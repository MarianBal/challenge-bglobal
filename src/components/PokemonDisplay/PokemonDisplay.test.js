import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import PokemonDisplay from './PokemonDisplay';

configure({ adapter: new Adapter() });

describe('PokemonDisplay renders correctly', () => {
  it('PokemonDisplay renders', () => {
    shallow(<PokemonDisplay />);
  });

  it('PokemonDisplay renders correctly', () => {
    const wrapper = shallow(<PokemonDisplay />);
    expect(wrapper).toMatchSnapshot();
  });
});
