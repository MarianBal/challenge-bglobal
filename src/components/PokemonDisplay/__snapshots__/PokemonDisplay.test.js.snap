// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDisplay renders correctly PokemonDisplay renders correctly 1`] = `ShallowWrapper {}`;
